include/RAJA/exec-openmp/forall_openmp.hxx

```diff
@@ -840,30 +244,11 @@ RAJA_INLINE void forall(
     IndexSetSegInfo* seg_info = ncis.getSegmentInfo(isi);
     DepGraphNode* task = seg_info->getDepGraphNode();
 
-    //
-    // This is declared volatile to prevent compiler from
-    // optimizing the while loop (into an if-statement, for example).
-    // It may not be able to see that the value accessed through
-    // the method call will be changed at the end of the for-loop
-    // from another executing thread.
-    //
-    volatile int* __restrict__ semVal = &(task->semaphoreValue());
-
-    while (*semVal != 0) {
-      /* spin or (better) sleep here */;
-      // printf("%d ", *semVal) ;
-      // sleep(1) ;
-      // for (volatile int spin = 0; spin<1000; ++spin) {
-      //    spin = spin ;
-      // }
-      sched_yield();
-    }
+    task->wait();
```

Comment:
That's a good point, it's definitely somewhere we could use more coverage. I've been testing this stuff with lulesh so far, and it seems ok at least for that for what that's worth.